src/NuGet.Core/NuGet.Protocol/Model/LocalPackageSearchMetadata.cs

```diff
@@ -70,6 +70,8 @@ public string Tags
 
         public string Title => !string.IsNullOrEmpty(_nuspec.GetTitle()) ? _nuspec.GetTitle() : _nuspec.GetId();
 
+        public LocalPackageInfo LocalPackageInfo => _package;
```

Comment:
Do we need LocalPackageInfo for anything beyond getting the PackageArchiveReader? 
Can we only add PackageArchiveReader to this API? 
If initialization is a concern we can make it lazy.

Reply:
I exposed `LocalPackageInfo` because the reader is a function that generates a Reader each time it is called. Then, I only exposed the PackageArchiveReader only when I need it. Making it lazy sounds a good idea.

Reply:
I changed the property to a Lazy PackageReaderBase.